CMakeLists.txt

```diff
@@ -28,6 +29,19 @@ set(LLVM_VERSION "${LLVM_VERSION_MAJOR}${LLVM_VERSION_MINOR}")
 file(TO_NATIVE_PATH "${LLVM_TOOLS_BINARY_DIR}/llvm-as${CMAKE_EXECUTABLE_SUFFIX}" LLVM_AS)
 file(TO_NATIVE_PATH "${LLVM_TOOLS_BINARY_DIR}/llvm-nm${CMAKE_EXECUTABLE_SUFFIX}" LLVM_NM)
 file(TO_NATIVE_PATH "${LLVM_TOOLS_BINARY_DIR}/clang${CMAKE_EXECUTABLE_SUFFIX}" CLANG)
+file(TO_NATIVE_PATH "${LLVM_TOOLS_BINARY_DIR}/llvm-config${CMAKE_EXECUTABLE_SUFFIX}" LLVM_CONFIG)
+
+# LLVM doesn't appear to expose --system-libs via its CMake interface, 
+# so we must shell out to llvm-config to find this info
+execute_process(COMMAND ${LLVM_CONFIG} --system-libs OUTPUT_VARIABLE HALIDE_SYSTEM_LIBS)
+string(STRIP "${HALIDE_SYSTEM_LIBS}" HALIDE_SYSTEM_LIBS)  # strip whitespace from start & end
+string(REPLACE " " ";" HALIDE_SYSTEM_LIBS "${HALIDE_SYSTEM_LIBS}")  # convert into a list
+if("${HALIDE_SYSTEM_LIBS}" STREQUAL "")
+  # It's theoretically possible that this could be legitimately empty,
+  # but in practice that doesn't really happen, so we'll assume it means we
+  # aren't configured correctly.
+  message(FATAL_ERROR "llvm-config --system-libs is empty; this is probably wrong.")
```

Comment:
Might be clearer with single quotes around the command itself ('llvm-config --system-libs').

Reply:
Done